### DISCUSSION TITLE
Ties of association are being made in the wrong direction

### DISCUSSION THREAD

**Discussion by nullptroma** [gaphor/gaphor#3553] [Q&A]
Ties of association are being made in the wrong direction. I use the automatic class diagram layout (Ctrl+Shift+L) and it should usually, as I understand it, place dependencies from bottom to top. So the classes at the bottom are very dependent and the classes above are independent, right? But for associations, the automatic arrangement is not the same as the dependency direction. I attach an example in the screenshot. The car is engine dependent, but the automatic arrangement has placed the engine lower, why? Is this a bug or am I missing something?

Translated with DeepL.com (free version)

![image](https://github.com/user-attachments/assets/ec6155b2-0929-4255-abbf-b16ff4967140)

**Comment by danyeaw** [comment]
Hi @nullptroma, for a composition association, normally you have one element at the top and many elements below. Like your example, an engine is part of a car. Here is a more complete example of a car showing the same type of layout:

![car-example-bdd-ibd-1802364589.png](https://github.com/user-attachments/assets/c75219ee-67af-4b4e-aaf1-7c62262fbfa3)

**Reply by nullptroma** [reply]
@danyeaw 
First of all, thank you for your attention
In general, you are right, it looks logical and convenient, but only if there are no other entities around.
Here are two examples. On one I put the dependencies backwards and then manually changed the direction of the arrows, and the other is the default. The default one is much harder to read and perceive, as the dependencies are not sorted from bottom to top. What if this is how I use it: add dependencies in the wrong direction and then manually change the arrows? Seems easier to read the diagram that way, less overlap
![Domain From the bottom to the top-1](https://github.com/user-attachments/assets/3d8eba0e-d35c-4857-9950-54e5da0b4165)
![Domain_??-1](https://github.com/user-attachments/assets/2571d514-2c58-4764-befb-ac88fdf04875)
